src/Accounts/Accounts/CommonModule/TelemetryProvider.cs

```diff
@@ -186,6 +176,35 @@ public virtual AzurePSQoSEvent CreateQosEvent(InvocationInfo invocationInfo, str
             return qosEvent;
         }
 
+        private const string privateAssemblySuffix = ".private";
+        private const string privateAssemblyPrefix = "Az.";
+        private static readonly int privateAssemblyPrefixLength = privateAssemblyPrefix.Length;
+        /// <summary>
+        /// below is workaround that current invocationInfo only contains private module name. Trimming '.private' is a workaround for the time being.
+        /// </summary>
+        /// <param name="moduleName"></param>
+        /// <returns></returns>
+        internal static string TrimModuleName(string moduleName)
+        {
+            if (moduleName != null && moduleName.StartsWith(privateAssemblyPrefix) && moduleName.EndsWith(privateAssemblySuffix))
+            {
+                int splitter = moduleName.Substring(privateAssemblyPrefixLength).IndexOf('.');
+                moduleName = moduleName.Substring(0, splitter + privateAssemblyPrefixLength);
```

Comment:
```suggestion
                moduleName = moduleName.Substring(0, moduleName.LastIndexOf('.'));
```

Reply:
cmdlet tries to fetch the second period character. Original logic should be correct.